### ISSUE TITLE
Question: IndexRoute with reducer?

### ISSUE THREAD

**Issue by albertovilva** [davezuko/react-redux-starter-kit#1117]
Hello,

I have the following set up for my routes.

<img width="319" alt="screen shot 2016-12-22 at 21 04 05" src="https://cloud.githubusercontent.com/assets/8507571/21438648/61234ac8-c88a-11e6-96a5-5b830cf76e3b.png">

Everything works great, Errors my Public area (with it childs if necessary) and the Private area with its childs.

The problem is the Index of "Private" as I can't get to work a definition similar to the rest of routes. As I see the sample ones they are defined like this:

<img width="389" alt="screen shot 2016-12-22 at 21 06 22" src="https://cloud.githubusercontent.com/assets/8507571/21438734/d5495b7c-c88a-11e6-8964-18646cefbbb5.png">

But if I want to use a definition like:

<img width="559" alt="screen shot 2016-12-22 at 21 09 07" src="https://cloud.githubusercontent.com/assets/8507571/21438752/eaeb9b02-c88a-11e6-9e09-98ca7118da10.png">

Basically I want my IndexRoute to behave like the rest of childRoutes. Im pretty sure Im not understanding how the routing fully works in this proyect, so any insight is more than appreciated.

Thanks and Merry Christmas!

**Comment by yupan21** [comment]
Your indexRoute is the "Public" sync route just with “PublicView” Component. You could change indexRoute into a function with store param the same as its childRoutes :
`indexRoute  : PublicRoute(store),`

PublicRoute removed require.ensure and has injectReducer with its own key:

import PublicView from './components/PublicView'
import { injectReducer } from '../../store/reducers'

export default (store) => ({
  getComponent (nextState, cb) {
      const Rate = require('./containers/PublicContainer').default
      const reducer = require('./modules/Public').default
      injectReducer(store, { key: 'public', reducer })

      /*  Return getComponent   */
      cb(null, Rate)
  }
})

**Comment by albertovilva** [comment]
@yupan21 Thank you very much. I tried exactly that without success, but now trying again it works like a charm! :D
